Path as setting

julesvirallinen opened the pull request:
relates to #193

Allows passing worklets paths via options to client, allowing for secure CSPs. 

Should have worklet files as js in a specific folder to allow automatic updating on CI, so that we can suggest something like 

```
import { viteStaticCopy } from 'vite-plugin-static-copy'
import { createRequire } from 'node:module';

const require = createRequire(import.meta.url);

export default {
  plugins: [
    viteStaticCopy({
      targets: [
        {
          src: require.resolve('@elevenlabs/client')/dist/worklets/audio-concat-processor.js',
          dest: 'dist',
        },
        {
          src: require.resolve('@elevenlabs/client')/dist/worklets/raw-audio-processor.js',
          dest: 'dist',
        },
      ],
    }),
  ],
}
```
which would automatically keep the files up-to-date (thanks @aapzu)

julesvirallinen committed:
add input props for worker path override

julesvirallinen committed:
feat: allow passing custom path for worklets

julesvirallinen committed:
add worklet path readme

julesvirallinen committed:
add build script to create scripts

aapzu commented on packages/react/README.md:
```diff
@@ -491,6 +491,57 @@ const { canSendFeedback } = useConversation();
+    viteStaticCopy({
+      targets: [
+        {
+          src: require.resolve('@elevenlabs/client')/dist/worklets/audio-concat-processor.js',
```
```suggestion
          src: path.join(require.resolve('@elevenlabs/client'), 'dist/worklets/audio-concat-processor.js'),
```

aapzu commented on packages/react/README.md:
```diff
@@ -491,6 +491,57 @@ const { canSendFeedback } = useConversation();
+          dest: 'dist',
+        },
+        {
+          src: require.resolve('@elevenlabs/client')/dist/worklets/raw-audio-processor.js',
```
```suggestion
          src: path.join(require.resolve('@elevenlabs/client'), 'dist/worklets/raw-audio-processor.js'),
```

aapzu commented on packages/client/README.md:
```diff
@@ -429,6 +429,46 @@ await conversation.changeOutputDevice({
+    viteStaticCopy({
+      targets: [
+        {
+          src: require.resolve('@elevenlabs/client')/dist/worklets/audio-concat-processor.js',
```
```suggestion
          src: path.join(require.resolve('@elevenlabs/client'), 'dist/worklets/audio-concat-processor.js'),
```

aapzu commented on packages/client/README.md:
```diff
@@ -429,6 +429,46 @@ await conversation.changeOutputDevice({
+          dest: 'dist',
+        },
+        {
+          src: require.resolve('@elevenlabs/client')/dist/worklets/raw-audio-processor.js',
```
```suggestion
          src: path.join(require.resolve('@elevenlabs/client'), 'dist/worklets/raw-audio-processor.js'),
```

AngeloGiacco committed:
fix powered by (#198)

powered by

PaulAsjes committed:
Widget support signed url (#197)

* Widget support for authed agents via signed-url

Reads and uses agent_id and conversation_signature
values from the signed-url when fetching config
and starting conversation.

* Renaming to resolvedAgentId

* Renaming to currentAgentId

* fix timeout bug

---------

Co-authored-by: Phillip Markert <phillip@ephisys.com>

PaulAsjes committed:
[convai-cli] Bump js lib version (#201)

Bump js lib version

PaulAsjes committed:
Add correction and mcp tool call events

PaulAsjes committed:
Also bump this

PaulAsjes committed:
export types not interfaces

AngeloGiacco committed:
fix cli react v (#192)

louisjoecodes committed:
chore: migrate to turbo

louisjoecodes committed:
feat: asyncapi type generation

louisjoecodes committed:
chore: add README

louisjoecodes committed:
fix: type generation

louisjoecodes committed:
fix: types

louisjoecodes committed:
chore: complete all types & fix generated types

AngeloGiacco committed:
[convai cli] start (#177)

* fix start/help cli view

* fix start/help cli view

* fix

* fix

AngeloGiacco committed:
powered by elevenlabs agents (#207)

powered by

louisjoecodes committed:
Map client SDK types to newly generated events (#213)

* fix: type generation

* fix: types

* chore: complete all types & fix generated types

* fix: map types to generated

* chore: ignore TS error in HelpView

AngeloGiacco committed:
[convai cli] fix lint (#215)

* powered by

* Add HelpView component

* tsconfig

* readd type annotation

louisjoecodes committed:
chore: add missing callbacks to client (#216)

* chore: add missing callbacks to client

* remove unused dependency

louisjoecodes committed:
chore: complete callbacks (#218)

louisjoecodes committed:
Bring react native up to speed with react SDK & port to generated tyes (#220)

* Bring react native up to speed with react SDK & port to generated tyes

* chore: remove comment

louisjoecodes committed:
[#210] - Fix stale closure on mic & mute ref (#222)

fix: type generation

louisjoecodes committed:
chore: add error messages to asyncapi (#223)

louisjoecodes committed:
chore: update README (#221)

* chore: update README

* chore: update README

* chore: tie together package.json commands

* chore: remove EAS bootstrap config

* chore: small tweaks

louisjoecodes committed:
chore: make error message optional (#226)

louisjoecodes committed:
chore: add publishconfig (#230)

AngeloGiacco committed:
[react] set DR for livekit (#229)

fix DR for livekit

AngeloGiacco committed:
[cli] fix header conversion (#232)

fix header conversion

AngeloGiacco committed:
[agents cli] convai cli -> agents cli  (#217)

* powered by

* rename convai-cli to agents-cli

* more rename

* more fix

* fix

* fix

* lockfile

AngeloGiacco committed:
[agents] cli - testing support (#233)

* testing

* fix

* test tests

* test lock test

* test updates

* redo test env

* fix test

* fix all lint issues

* fix test

* almost all solved

* fix test

* fix lints

* fix

AngeloGiacco committed:
[agents cli] fetch tools (#235)

* fetch tools

* fetch-tools

* lint fixes

AngeloGiacco committed:
[agents cli] add sync tools (#236)

* add sync-tools command

* add tests for sync-tools

* add sync-tools ui view

* intiial fix

* fix tests

* fix test

* fix tests]

* fix lint

* bump version

BrendonSCooblue committed:
[agents cli] update API base URLs for residency endpoints (#234)

fix: update API base URLs for residency endpoints

AngeloGiacco committed:
bump (#238)

PaulAsjes committed:
[Widget] Add expand event (#243)

* Add expand event

* bump version

julesvirallinen committed:
client: pr comments

julesvirallinen committed:
chore: bump versions
